helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1299,6 +1308,10 @@ protected void checkLiveInstancesObservation(List<LiveInstance> liveInstances,
   }
 
   public void shutdown() throws InterruptedException {
+    if (_periodicRebalancerFutureTasks != null) {
+      _periodicRebalancerFutureTasks.cancel(false);
+    }
```

Comment:
Why need to cancel the task while stopPeriodRebalance is called here?

Reply:
Yea you are correct. shutDown will cancel tasks later. Updated.